64-blue-whale/technical_specification.md

```diff
@@ -0,0 +1,130 @@
+- Full test coverage
+
+### Not included:
+
+- Automated end-to-end tests against the real backend
+
+## Details:
+
+This section outlines a checklist of all features to be implemented, divided into several steps. A pre-release should be published upon the completion of each step. Final releases should begin with version 2 to distinguish them from the legacy frontend, which uses version numbers in the version 1 range.
+
+### Already implemented
+
+The following parts are already contained in the baseline application that has been created in the [Hippopotamus](../60-hippopotamus/technical_specification.md) epic.
+
+- Homepage
+	- Global Stats
+- Header and Footer
+- Dataset browser (partial)
+- Auth Service
+- Error message handling
+- User Login:
+	- Login/Logout buttons
+	- OIDC callback URL
+	- User registration
+	- 2FA setup
+	- 2FA confirmation
+- Metadata service
+- Metadata search service (partial)
+
+We have also added rudimentary unit tests and end-to-end tests. While implementing the remaining functionality in the following four steps, basic tests should be included together with the implemented functionality. As a last step, the implementation and tests should be refined.
+
+### Step 1: Catalog functionality
+
+In this epic, we first implement functions from the GHGA Catalog.
+
+- [ ] Metadata search service (complete)
+- [ ] Dataset browser (complete)
+	- [ ] Result list
+	- [ ] Dataset summaries
+	- [ ] Filter (facets)
+	- [ ] Search
+ - [ ] Single dataset view
+	 - [ ] Summary
+	 - [ ] Study
+	 - [ ] Publications
+	 - [ ] Dataset access policy
+- [ ] Access Request Service
+- [ ] Access request
+	 - [ ] Button to request access
+	 - [ ] Access request form
+
+## Step 2: Access Requests
+
+We implement the functionality of the account page in the next step:
+
+- [ ] IVA service
+- [ ] Account (user profile) page
+	- [ ] Email address
+	- [ ] Contact addresses (IVAs)
+		- [ ] Create
+		- [ ] Request verification
+		- [ ] Confirm verification code
+		- [ ] Delete
+	- [ ] Accessible datasets
+	- [ ] Pending access requests
+
+## Step 3: Download
+
+We complete the user facing functionality by implementing the function to create a work package:
+
+- [ ] Work packages
+    - [ ] Work package service
+	- [ ] Creation form
+	- [ ] Token generation
+
+## Step 4: Data steward tools
+
+Finally, we add the tooling for the data stewards:
+
+- [ ] Access request manager
+	- [ ] List
+	- [ ] Filter
+	- [ ] Detail
+- [ ] IVA manager
+	- [ ] List
+	- [ ] Filter
+	- [ ] (Re)create code
+	- [ ] Invalidation
+	- [ ] Confirm transmission
+
+### Polish and improve test coverage
+
+In the last step, we polish everything (edge cases, layout etc.) and increase the test coverage from unit tests and end-to-end tests. The latter should cover the complete user journey from dataset discovery to dataset download, but they should use the static (stateless) mock backend that is part of the frontend repository.
```

Comment:
We may perhaps want to include small-screen responsiveness too in this section, which might take half a sprint for all the pages and trying to deal with tailwind and it not overriding material styles.